dist/src/main/java/io/camunda/zeebe/shared/management/ClusterEndpoint.java

```diff
@@ -361,6 +362,16 @@ public ResponseEntity<?> removeSubResource(
     };
   }
 
+  @PostMapping(path = "/purge", produces = "application/json")
+  public ResponseEntity<?> purge(@RequestParam(defaultValue = "false") final boolean dryRun) {
+    try {
+      return ClusterApiUtils.mapOperationResponse(
+          requestSender.purge(new PurgeRequest(dryRun)).join());
```

Comment:
:upside_down_face: I know we do this every where in this controller, but it always makes me twitch to see a `join` on a future :smile: 

I think the following should just work OOTB - if it doesn't, then ignore this comment. But otherwise, let's let Spring deal with the asynchronicity:

```java
  @PostMapping(path = "/purge", produces = "application/json")
  public CompletableFuture<ResponseEntity<?>> purge(
      @RequestParam(defaultValue = "false") final boolean dryRun) {
    try {
      requestSender
          .purge(new PurgeRequest(dryRun))
          .thenApply(ClusterApiUtils::mapOperationResponse)
          .exceptionally(ClusterApiUtils::mapError);
    } catch (final Exception error) {
      return ClusterApiUtils.mapError(error);
    }
  }
```

Reply:
Spring can handle it, mvn says it's fine, but my IDE complains, but I'll just ignore the whining:
![image](https://github.com/user-attachments/assets/534c8f19-680f-477b-beda-ad19e4c0990c)

Latest Intellij, did restart, maybe it will vanish eventually.

Reply:
:grinning: The `cannot convert T to T` message just makes me think of

![image](https://github.com/user-attachments/assets/f852f903-934b-4353-801d-b6d08f73970b)